tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+  (which are always sub-leading).
+  Args:
+    tensors: list of the tensors in the network.
+    connects: length-N list of lists (or tuples) specifying the network 
+      connections. The jth entry of the ith list in connects labels the edge 
+      connected to the jth index of the ith tensor. Labels should be positive 
+      integers for internal indices and negative integers for free indices.
+    cont_order: optional argument to specify the order for contracting the 
+      positive indices. Defaults to ascending order if omitted. 
+  Returns:
+    float: the cost of the network contraction, given as log10(total_FLOPS).
+  """
+  
+  total_cost = None
+  num_tensors = len(tensor_list_in)
+  tensor_dims = [np.array(np.log10(tensor_list_in[ele].shape)) 
+                 for ele in range(num_tensors)]
+  connect_list = [np.array(connect_list_in[ele]) for ele in range(num_tensors)]
+  
+  # generate contraction order if necessary
+  flat_connect = np.concatenate(connect_list)
+  if cont_order is None:
+    cont_order = np.unique(flat_connect[flat_connect > 0])
+  else:
+    cont_order = np.array(cont_order)
+  
+  # do all partial traces
+  for ele in range(num_tensors):
+    uni_inds, counts = np.unique(connect_list[ele],return_counts=True)
+    tr_inds = uni_inds[counts==2]
+    temp_connect = connect_list[ele]
+    temp_dims = tensor_dims[ele]
+    for ind in tr_inds:
+      temp_dims = temp_dims[temp_connect != ind]
+      temp_connect = temp_connect[temp_connect != ind]
+      cont_order = np.delete(cont_order, np.flatnonzero(cont_order == ind))
+      
+    tensor_dims[ele] = temp_dims
+    connect_list[ele] = temp_connect
+  
+  # do all binary contractions
+  while len(cont_order) > 0:
+    # identify tensors to be contracted
+    cont_ind = cont_order[0]
+    locs = [ele for ele in range(len(connect_list)) 
+            if sum(connect_list[ele] == cont_ind) > 0]
+  
+    # do binary contraction
+    cont_many, A_cont, B_cont = np.intersect1d(connect_list[locs[0]], 
+                                               connect_list[locs[1]], 
+                                               assume_unique=True, 
+                                               return_indices=True)
+    A_dim = np.sum(tensor_dims[locs[0]])
+    B_dim = np.sum(tensor_dims[locs[1]]) 
+    cont_dim = np.sum((tensor_dims[locs[0]])[A_cont])
+    single_cost = A_dim + B_dim - cont_dim
+    if total_cost is None:
+      total_cost =  single_cost
+    else:
+      total_cost =  total_cost + np.log10(1+10**(single_cost-total_cost))
+    
+    tensor_dims.append(np.append(np.delete(tensor_dims[locs[0]], A_cont), 
+                                 np.delete(tensor_dims[locs[1]], B_cont)))
+    connect_list.append(np.append(np.delete(connect_list[locs[0]], A_cont), 
+                                  np.delete(connect_list[locs[1]], B_cont)))
+  
+    # remove contracted tensors from list and update cont_order
+    del tensor_dims[locs[1]]
```

Comment:
see comment above. Don't use `del`

Reply:
As far as I am aware `del` is the preferred option to remove an item from a list at a known index location. Is there another option? I could instead `pop` un-needed items from the lists, but I do not see the advantage. I am going by:
https://stackoverflow.com/questions/11520492/difference-between-del-remove-and-pop-on-lists/11520540

Reply:
I think you're right, let's keep it the way it is